internal/csot/csot.go

```diff
@@ -8,29 +8,134 @@ package csot
 
 import (
 	"context"
+	"runtime/debug"
 	"time"
 )
 
-type timeoutKey struct{}
+type withoutMaxTime struct{}
```

Comment:
Suggest removing `timeoutKey`, which currently supersets it's necessity which is to imply a way to determine that context has a 0 value for infinite retires. We should be more specific and label the context "withoutMaxTime", indicating that we will not set a maxTimeMS value on wire message.